src/Api/QueryParameterValidator/QueryParameterValidator.php

```diff
@@ -60,8 +61,14 @@ public function validateFilters(string $resourceClass, array $resourceFilters, a
             }
 
             foreach ($filter->getDescription($resourceClass) as $name => $data) {
-                foreach ($this->validators as $validator) {
-                    if ($errors = $validator->validate($name, $data, $queryParameters)) {
+                $collectionFormat = $this->getCollectionFormat($data);
+
+                // validate simple values
+                $errorList[] = $this->validate($name, $data, $queryParameters);
```

Comment:
First validation is expected to allow to validate missing requirements. Non scalar parameters may cause the `Array to string conversion`